posts/python-in-production-part-2-of-5.md

```diff
@@ -0,0 +1,166 @@
+
+# Installing Your Program as a Local Package
+
+How do you use packages? Well, you have to install them with pip first!  Let's do that now.  From the command line, navigate to the src directory and install your *local* package.
+```bash
+cd ..
+python3 -m pip install .
+```
+
+**Note:** make sure you type the ' . ' after install.  This tells pip to install any packages it finds in the current directory.  You should see something familiar, like this...
+```bash
+Processing /Users/lucas/PersonalProjects/sample-python-project/src
+  Preparing metadata (setup.py) ... done
+Using legacy 'setup.py install' for simple-api, since package 'wheel' is not installed.
+Installing collected packages: simple-api
+  Running setup.py install for simple-api ... done
+Successfully installed simple-api-1.0
+```
+
+Congratulations, you have just created a python package!!! The pure simplicity with which we have gotten here is one of the beautiful things about Python.  This package isn't available on PyPi or anything like that, but *it could be!*  The next step is to run our program and see it in action.
+
+From the src directory...
+```bash
+python3 simple_api
+INFO:     Started server process [4054]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+You should see the familiar FastAPI startup text, and be able to access your "hello world" endpoint served at the root location by clicking the URL.  **Notice** that we did not have to append "***.py***" to simple_api when we ran it.  That is because it is now a package.  Similarly to how when you import something, you do not write ***import datetime.py***.
+
+Another *cool trick* that has now been made possible, is running our program from the python interpreter.  Try this in your terminal.  Type ***python*** to start the interpreter.
+```bash
+>>> from simple_api.api import app
+>>> import uvicorn
+>>> uvicorn.run(app)
+INFO:     Started server process [4087]
+INFO:     Waiting for application startup.
+INFO:     Application startup complete.
+INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+WHOA! You just ran your newly packaged project straight from the interpreter, as if it was part of the standard library!
+
+One thing to note, at this point in time, you **do not** want to add your local package to the requirements.txt file.  Your package cannot be a dependency to itself!  If you are in the habit of running ***pip freeze > requirements.txt***, then you may need to go in and delete your package from this file.  In a later part of this series, we will create a separate virtual environment and ***requirements.txt*** file that will be used to run unit tests, integration tests, and build scripts.  Following the steps outlined here will make that future work much easier.
```

Comment:
its a style thing and certainly up to you, but i'd just use the single backticks for inline code for snippets and file names like `requirements.txt` and `pip freeze > requirements.txt`

Reply:
So thaaaat's how you make inline code snippets!  I will definitely update this.